// @contentful/forma-36-react-components/dist/components/ValidationMessage/ValidationMessage.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.ValidationMessage = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _react = _interopRequireWildcard(require("react"));

var _classnames = _interopRequireDefault(require("classnames"));

var _Icon = _interopRequireDefault(require("../Icon/Icon"));

var styles = {
  "ValidationMessage": "ValidationMessage__ValidationMessage___3_rEq",
  "ValidationMessage__icon": "ValidationMessage__ValidationMessage__icon___3HPCh",
  "ValidationMessage__text": "ValidationMessage__ValidationMessage__text___8FBj5"
};
var defaultProps = {
  testId: 'cf-ui-validation-message'
};

var ValidationMessage =
/*#__PURE__*/
function (_Component) {
  (0, _inherits2.default)(ValidationMessage, _Component);

  function ValidationMessage() {
    (0, _classCallCheck2.default)(this, ValidationMessage);
    return (0, _possibleConstructorReturn2.default)(this, (0, _getPrototypeOf2.default)(ValidationMessage).apply(this, arguments));
  }

  (0, _createClass2.default)(ValidationMessage, [{
    key: "render",
    value: function render() {
      var _this$props = this.props,
          className = _this$props.className,
          children = _this$props.children,
          testId = _this$props.testId,
          otherProps = (0, _objectWithoutProperties2.default)(_this$props, ["className", "children", "testId"]);
      var classNames = (0, _classnames.default)(styles['ValidationMessage'], className);
      return _react.default.createElement("div", (0, _extends2.default)({}, otherProps, {
        className: classNames,
        "data-test-id": testId
      }), _react.default.createElement(_Icon.default, {
        icon: "ErrorCircle",
        className: styles['ValidationMessage__icon'],
        color: "negative"
      }), _react.default.createElement("p", {
        className: styles['ValidationMessage__text']
      }, children));
    }
  }]);
  return ValidationMessage;
}(_react.Component);

exports.ValidationMessage = ValidationMessage;
ValidationMessage.displayName = "ValidationMessage";
(0, _defineProperty2.default)(ValidationMessage, "propTypes", {
  className: _propTypes.default.string,
  testId: _propTypes.default.string,
  style: _propTypes.default.any,
  children: _propTypes.default.node.isRequired
});
(0, _defineProperty2.default)(ValidationMessage, "defaultProps", defaultProps);
var _default = ValidationMessage;
exports.default = _default;
