// @contentful/forma-36-react-components/dist/components/Skeleton/SkeletonBodyText/SkeletonBodyText.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.SkeletonBodyText = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _react = _interopRequireWildcard(require("react"));

var _SkeletonText = _interopRequireDefault(require("../SkeletonText"));

var defaultProps = {
  numberOfLines: 2,
  offsetTop: 0,
  offsetLeft: 0,
  lineHeight: 16,
  marginBottom: 8
};

var SkeletonBodyText =
/*#__PURE__*/
function (_Component) {
  (0, _inherits2.default)(SkeletonBodyText, _Component);

  function SkeletonBodyText() {
    (0, _classCallCheck2.default)(this, SkeletonBodyText);
    return (0, _possibleConstructorReturn2.default)(this, (0, _getPrototypeOf2.default)(SkeletonBodyText).apply(this, arguments));
  }

  (0, _createClass2.default)(SkeletonBodyText, [{
    key: "render",
    value: function render() {
      return _react.default.createElement(_SkeletonText.default, this.props);
    }
  }]);
  return SkeletonBodyText;
}(_react.Component);

exports.SkeletonBodyText = SkeletonBodyText;
SkeletonBodyText.displayName = "SkeletonBodyText";
(0, _defineProperty2.default)(SkeletonBodyText, "defaultProps", defaultProps);
var _default = SkeletonBodyText;
exports.default = _default;
