// @contentful/forma-36-react-components/dist/components/TextField/TextField.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.TextField = void 0;

var _objectSpread2 = _interopRequireDefault(require("@babel/runtime/helpers/objectSpread"));

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf3 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _assertThisInitialized2 = _interopRequireDefault(require("@babel/runtime/helpers/assertThisInitialized"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _react = _interopRequireWildcard(require("react"));

var _classnames = _interopRequireDefault(require("classnames"));

var _ValidationMessage = _interopRequireDefault(require("../ValidationMessage"));

var _FormLabel = _interopRequireDefault(require("../FormLabel/FormLabel"));

var _HelpText = _interopRequireDefault(require("../HelpText"));

var _TextInput = _interopRequireDefault(require("../TextInput/TextInput"));

var _TextLink = _interopRequireDefault(require("../TextLink/TextLink"));

var _Textarea = _interopRequireDefault(require("../Textarea/Textarea"));

var styles = {
  "TextField": "TextField__TextField___Sf6eo",
  "TextField__label-wrapper": "TextField__TextField__label-wrapper___2-MJT",
  "TextField__label-link": "TextField__TextField__label-link___u94cw",
  "TextField__validation-message": "TextField__TextField__validation-message___1Idkl",
  "TextField__help-text": "TextField__TextField__help-text___p4rVf",
  "TextField__hints": "TextField__TextField__hints___3Di2P",
  "TextField__count": "TextField__TextField__count___jiTTs"
};
var defaultProps = {
  testId: 'cf-ui-text-field',
  textarea: false,
  required: false,
  countCharacters: false
};

var TextField =
/*#__PURE__*/
function (_Component) {
  (0, _inherits2.default)(TextField, _Component);

  function TextField() {
    var _getPrototypeOf2;

    var _this;

    (0, _classCallCheck2.default)(this, TextField);

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this = (0, _possibleConstructorReturn2.default)(this, (_getPrototypeOf2 = (0, _getPrototypeOf3.default)(TextField)).call.apply(_getPrototypeOf2, [this].concat(args)));
    (0, _defineProperty2.default)((0, _assertThisInitialized2.default)(_this), "state", {
      value: _this.props.value || '',
      initialValue: _this.props.value
    });
    (0, _defineProperty2.default)((0, _assertThisInitialized2.default)(_this), "handleOnChange", function (evt) {
      _this.setState({
        value: evt.target.value
      });

      if (_this.props.onChange) _this.props.onChange(evt);
    });
    return _this;
  }

  (0, _createClass2.default)(TextField, [{
    key: "render",
    value: function render() {
      var _this$props = this.props,
          validationMessage = _this$props.validationMessage,
          className = _this$props.className,
          textInputProps = _this$props.textInputProps,
          testId = _this$props.testId,
          formLabelProps = _this$props.formLabelProps,
          textLinkProps = _this$props.textLinkProps,
          labelText = _this$props.labelText,
          helpText = _this$props.helpText,
          textarea = _this$props.textarea,
          countCharacters = _this$props.countCharacters,
          required = _this$props.required,
          onChange = _this$props.onChange,
          onBlur = _this$props.onBlur,
          onFocus = _this$props.onFocus,
          value = _this$props.value,
          name = _this$props.name,
          id = _this$props.id,
          otherProps = (0, _objectWithoutProperties2.default)(_this$props, ["validationMessage", "className", "textInputProps", "testId", "formLabelProps", "textLinkProps", "labelText", "helpText", "textarea", "countCharacters", "required", "onChange", "onBlur", "onFocus", "value", "name", "id"]);
      var classNames = (0, _classnames.default)(styles['TextField'], className); // eslint-disable-next-line @typescript-eslint/no-explicit-any

      var Element = textarea ? _Textarea.default : _TextInput.default;
      return _react.default.createElement("div", (0, _extends2.default)({
        className: classNames
      }, otherProps, {
        "data-test-id": testId
      }), _react.default.createElement("div", {
        className: styles['TextField__label-wrapper']
      }, _react.default.createElement(_FormLabel.default, (0, _extends2.default)({}, formLabelProps, {
        htmlFor: id,
        required: required
      }), labelText), textLinkProps && _react.default.createElement(_TextLink.default, (0, _objectSpread2.default)({}, textLinkProps, {
        className: styles['TextField__label-link']
      }), textLinkProps.text)), _react.default.createElement(Element, (0, _objectSpread2.default)({
        error: !!validationMessage,
        name: name,
        id: id,
        onBlur: onBlur,
        onFocus: onFocus,
        onChange: this.handleOnChange,
        value: value,
        required: required
      }, textInputProps)), validationMessage && _react.default.createElement(_ValidationMessage.default, {
        className: styles['TextField__validation-message']
      }, validationMessage), (helpText || countCharacters) && _react.default.createElement("div", {
        className: styles['TextField__hints']
      }, helpText && _react.default.createElement(_HelpText.default, {
        className: styles['TextField__help-text']
      }, helpText), countCharacters && textInputProps && textInputProps.maxLength && _react.default.createElement(_HelpText.default, {
        className: (0, _classnames.default)(styles['TextField__help-text'], styles['TextField__count'])
      }, this.state.value ? this.state.value.length : 0, "/", textInputProps.maxLength)));
    }
  }], [{
    key: "getDerivedStateFromProps",
    value: function getDerivedStateFromProps(props, state) {
      if (props.value !== state.initialValue) {
        return (0, _objectSpread2.default)({}, state, {
          value: props.value,
          initialValue: props.value
        });
      }

      return state;
    }
  }]);
  return TextField;
}(_react.Component);

exports.TextField = TextField;
TextField.displayName = "TextField";
(0, _defineProperty2.default)(TextField, "propTypes", {
  name: _propTypes.default.string.isRequired,
  id: _propTypes.default.string.isRequired,
  labelText: _propTypes.default.string.isRequired,
  value: _propTypes.default.string,
  validationMessage: _propTypes.default.string,
  testId: _propTypes.default.string,
  className: _propTypes.default.string,
  formLabelProps: _propTypes.default.any,
  textLinkProps: _propTypes.default.any,
  textInputProps: _propTypes.default.oneOfType([_propTypes.default.any, _propTypes.default.any]),
  helpText: _propTypes.default.string,
  required: _propTypes.default.bool,
  textarea: _propTypes.default.bool,
  countCharacters: _propTypes.default.bool,
  onChange: _propTypes.default.func,
  onBlur: _propTypes.default.func,
  onFocus: _propTypes.default.func
});
(0, _defineProperty2.default)(TextField, "defaultProps", defaultProps);
var _default = TextField;
exports.default = _default;
