// @contentful/forma-36-react-components/dist/components/Asset/Asset.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.Asset = exports.types = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf3 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _assertThisInitialized2 = _interopRequireDefault(require("@babel/runtime/helpers/assertThisInitialized"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _react = _interopRequireWildcard(require("react"));

var _classnames = _interopRequireDefault(require("classnames"));

var _Illustration = _interopRequireDefault(require("../Illustration/Illustration"));

var styles = {
  "Asset": "Asset__Asset___1zgnB",
  "Asset__image-container": "Asset__Asset__image-container___1oHDv",
  "Asset__image-container__image": "Asset__Asset__image-container__image___3J2Ik",
  "Asset__title-container": "Asset__Asset__title-container___jUj2R",
  "Asset__title-container__title": "Asset__Asset__title-container__title___1AHiK",
  "Asset__asset-container": "Asset__Asset__asset-container___226So",
  "Asset__asset-container__title": "Asset__Asset__asset-container__title___y3fAq",
  "Asset__illustration-container": "Asset__Asset__illustration-container___D0AVE"
};
var types = {
  archive: 'archive',
  audio: 'audio',
  code: 'code',
  image: 'image',
  markup: 'markup',
  pdf: 'pdf',
  plaintext: 'plaintext',
  presentation: 'presentation',
  richtext: 'richtext',
  spreadsheet: 'spreadsheet',
  video: 'video'
};
exports.types = types;
var defaultProps = {
  type: 'image',
  testId: 'cf-ui-asset'
};

var Asset =
/*#__PURE__*/
function (_Component) {
  (0, _inherits2.default)(Asset, _Component);

  function Asset() {
    var _getPrototypeOf2;

    var _this;

    (0, _classCallCheck2.default)(this, Asset);

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this = (0, _possibleConstructorReturn2.default)(this, (_getPrototypeOf2 = (0, _getPrototypeOf3.default)(Asset)).call.apply(_getPrototypeOf2, [this].concat(args)));
    (0, _defineProperty2.default)((0, _assertThisInitialized2.default)(_this), "renderImage", function (src, title) {
      return _react.default.createElement(_react.default.Fragment, null, _react.default.createElement("div", {
        className: styles['Asset__image-container']
      }, _react.default.createElement("img", {
        className: styles['Asset__image-container__image'],
        src: src,
        alt: title
      })), _react.default.createElement("div", {
        className: styles['Asset__title-container']
      }, _react.default.createElement("span", {
        className: styles['Asset__title-container__title']
      }, title)));
    });
    (0, _defineProperty2.default)((0, _assertThisInitialized2.default)(_this), "renderAsset", function (type, title) {
      var illustraionName = type.charAt(0).toUpperCase() + type.slice(1);
      return _react.default.createElement("div", {
        className: styles['Asset__asset-container']
      }, _react.default.createElement("div", {
        className: styles['Asset__illustration-container']
      }, _react.default.createElement(_Illustration.default, {
        illustration: illustraionName
      })), _react.default.createElement("span", {
        className: styles['Asset__asset-container__title']
      }, title));
    });
    return _this;
  }

  (0, _createClass2.default)(Asset, [{
    key: "render",
    value: function render() {
      var _this$props = this.props,
          className = _this$props.className,
          src = _this$props.src,
          title = _this$props.title,
          type = _this$props.type,
          testId = _this$props.testId,
          otherProps = (0, _objectWithoutProperties2.default)(_this$props, ["className", "src", "title", "type", "testId"]);
      var classNames = (0, _classnames.default)(styles.Asset, className);
      return _react.default.createElement("div", (0, _extends2.default)({
        className: classNames,
        "data-test-id": testId
      }, otherProps), type && type === 'image' ? this.renderImage(src, title) : this.renderAsset(type, title));
    }
  }]);
  return Asset;
}(_react.Component);

exports.Asset = Asset;
Asset.displayName = "Asset";
(0, _defineProperty2.default)(Asset, "propTypes", {
  src: _propTypes.default.string.isRequired,
  title: _propTypes.default.string.isRequired,
  type: _propTypes.default.any,
  className: _propTypes.default.string,
  testId: _propTypes.default.string
});
(0, _defineProperty2.default)(Asset, "defaultProps", defaultProps);
var _default = Asset;
exports.default = _default;
