// @contentful/forma-36-react-components/dist/components/Select/Select/Select.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.Select = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf3 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _assertThisInitialized2 = _interopRequireDefault(require("@babel/runtime/helpers/assertThisInitialized"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _react = _interopRequireWildcard(require("react"));

var _classnames = _interopRequireDefault(require("classnames"));

var _Icon = _interopRequireDefault(require("../../Icon/Icon"));

var styles = {
  "Select__wrapper": "Select__Select__wrapper___2pBIz",
  "Select": "Select__Select___31Z46 a11y__focus-border--default___60AXp",
  "Select--disabled": "Select__Select--disabled___3n58N",
  "Select--small": "Select__Select--small___1C0xZ",
  "Select--medium": "Select__Select--medium___1Hmlq",
  "Select--large": "Select__Select--large___1PgJW",
  "Select--full": "Select__Select--full___7ei7d",
  "Select--auto": "Select__Select--auto___42_yz",
  "Select--negative": "Select__Select--negative___3nxwU",
  "Select__icon": "Select__Select__icon___boRAZ"
};
var defaultProps = {
  testId: 'cf-ui-select',
  required: false,
  hasError: false,
  isDisabled: false,
  width: 'full',
  willBlurOnEsc: true
};

var Select =
/*#__PURE__*/
function (_Component) {
  (0, _inherits2.default)(Select, _Component);

  function Select() {
    var _getPrototypeOf2;

    var _this;

    (0, _classCallCheck2.default)(this, Select);

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this = (0, _possibleConstructorReturn2.default)(this, (_getPrototypeOf2 = (0, _getPrototypeOf3.default)(Select)).call.apply(_getPrototypeOf2, [this].concat(args)));
    (0, _defineProperty2.default)((0, _assertThisInitialized2.default)(_this), "state", {
      value: _this.props.value
    });
    (0, _defineProperty2.default)((0, _assertThisInitialized2.default)(_this), "handleKeyDown", function (e) {
      var ESC = 27;

      if (e.keyCode === ESC && _this.props.willBlurOnEsc) {
        e.currentTarget.blur();
      }
    });
    return _this;
  }

  (0, _createClass2.default)(Select, [{
    key: "UNSAFE_componentWillReceiveProps",
    value: function UNSAFE_componentWillReceiveProps(nextProps) {
      if (this.props.value !== nextProps.value) {
        this.setState({
          value: nextProps.value
        });
      }
    }
  }, {
    key: "render",
    value: function render() {
      var _cn,
          _this2 = this;

      var _this$props = this.props,
          id = _this$props.id,
          name = _this$props.name,
          required = _this$props.required,
          children = _this$props.children,
          width = _this$props.width,
          className = _this$props.className,
          testId = _this$props.testId,
          onChange = _this$props.onChange,
          onBlur = _this$props.onBlur,
          onFocus = _this$props.onFocus,
          isDisabled = _this$props.isDisabled,
          hasError = _this$props.hasError,
          willBlurOnEsc = _this$props.willBlurOnEsc,
          otherProps = (0, _objectWithoutProperties2.default)(_this$props, ["id", "name", "required", "children", "width", "className", "testId", "onChange", "onBlur", "onFocus", "isDisabled", "hasError", "willBlurOnEsc"]);
      var widthClass = "Select--".concat(width);
      var classNames = (0, _classnames.default)(styles['Select'], (_cn = {}, (0, _defineProperty2.default)(_cn, styles['Select--disabled'], isDisabled), (0, _defineProperty2.default)(_cn, styles['Select--negative'], hasError), _cn));
      var wrapperClassNames = (0, _classnames.default)(styles['Select__wrapper'], styles[widthClass], className);
      return _react.default.createElement("div", {
        className: wrapperClassNames
      }, _react.default.createElement("select", (0, _extends2.default)({
        id: id,
        required: required,
        name: name,
        "aria-label": name,
        "data-test-id": testId,
        className: classNames,
        value: this.state.value,
        disabled: isDisabled,
        onFocus: onFocus,
        onChange: function (_onChange) {
          function onChange(_x) {
            return _onChange.apply(this, arguments);
          }

          onChange.toString = function () {
            return _onChange.toString();
          };

          return onChange;
        }(function (e) {
          if (!isDisabled) {
            _this2.setState({
              value: e.target.value
            });

            if (onChange) {
              onChange(e);
            }
          }
        }),
        onBlur: onBlur,
        onKeyDown: this.handleKeyDown
      }, otherProps), children), _react.default.createElement(_Icon.default, {
        className: styles['Select__icon'],
        icon: "ArrowDown",
        color: "muted"
      }));
    }
  }]);
  return Select;
}(_react.Component);

exports.Select = Select;
Select.displayName = "Select";
(0, _defineProperty2.default)(Select, "propTypes", {
  required: _propTypes.default.bool,
  name: _propTypes.default.string,
  id: _propTypes.default.string,
  hasError: _propTypes.default.bool,
  value: _propTypes.default.string,
  isDisabled: _propTypes.default.bool,
  onChange: _propTypes.default.func,
  onBlur: _propTypes.default.func,
  onFocus: _propTypes.default.func,
  width: _propTypes.default.oneOf(['auto', 'small', 'medium', 'large', 'full']),
  testId: _propTypes.default.string,
  className: _propTypes.default.string,
  children: _propTypes.default.node.isRequired,
  willBlurOnEsc: _propTypes.default.bool.isRequired
});
(0, _defineProperty2.default)(Select, "defaultProps", defaultProps);
var _default = Select;
exports.default = _default;
