// @contentful/forma-36-react-components/dist/components/RadioButton/RadioButton.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.RadioButton = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _react = _interopRequireWildcard(require("react"));

var _ControlledInput = _interopRequireDefault(require("../ControlledInput/ControlledInput"));

var defaultProps = {
  required: false,
  disabled: false,
  type: 'radio',
  testId: 'cf-ui-radio-button',
  willBlurOnEsc: true
};

var RadioButton =
/*#__PURE__*/
function (_Component) {
  (0, _inherits2.default)(RadioButton, _Component);

  function RadioButton() {
    (0, _classCallCheck2.default)(this, RadioButton);
    return (0, _possibleConstructorReturn2.default)(this, (0, _getPrototypeOf2.default)(RadioButton).apply(this, arguments));
  }

  (0, _createClass2.default)(RadioButton, [{
    key: "render",
    value: function render() {
      return _react.default.createElement(_ControlledInput.default, this.props);
    }
  }]);
  return RadioButton;
}(_react.Component);

exports.RadioButton = RadioButton;
RadioButton.displayName = "RadioButton";
(0, _defineProperty2.default)(RadioButton, "defaultProps", defaultProps);
RadioButton.defaultProps = defaultProps;
var _default = RadioButton;
exports.default = _default;
