// @contentful/forma-36-react-components/dist/components/Dropdown/DropdownList/DropdownList.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.DropdownList = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _react = _interopRequireWildcard(require("react"));

var _classnames = _interopRequireDefault(require("classnames"));

var styles = {
  "DropdownList": "DropdownList__DropdownList___2EMLM",
  "DropdownList--border-top": "DropdownList__DropdownList--border-top___ojlyY",
  "DropdownList--border-bottom": "DropdownList__DropdownList--border-bottom___3F_82"
};
var defaultProps = {
  testId: 'cf-ui-dropdown-list'
};

var DropdownList =
/*#__PURE__*/
function (_Component) {
  (0, _inherits2.default)(DropdownList, _Component);

  function DropdownList() {
    (0, _classCallCheck2.default)(this, DropdownList);
    return (0, _possibleConstructorReturn2.default)(this, (0, _getPrototypeOf2.default)(DropdownList).apply(this, arguments));
  }

  (0, _createClass2.default)(DropdownList, [{
    key: "render",
    value: function render() {
      var _this$props = this.props,
          className = _this$props.className,
          border = _this$props.border,
          maxHeight = _this$props.maxHeight,
          testId = _this$props.testId,
          children = _this$props.children,
          listRef = _this$props.listRef,
          otherProps = (0, _objectWithoutProperties2.default)(_this$props, ["className", "border", "maxHeight", "testId", "children", "listRef"]);
      var classNames = (0, _classnames.default)(styles['DropdownList'], className, (0, _defineProperty2.default)({}, styles["DropdownList--border-".concat(border)], border));
      return _react.default.createElement("ul", (0, _extends2.default)({
        ref: listRef,
        "data-test-id": testId,
        role: "menu",
        style: {
          maxHeight: maxHeight || 'auto',
          overflowY: maxHeight ? 'auto' : 'visible'
        },
        className: classNames
      }, otherProps), children);
    }
  }]);
  return DropdownList;
}(_react.Component);

exports.DropdownList = DropdownList;
DropdownList.displayName = "DropdownList";
(0, _defineProperty2.default)(DropdownList, "propTypes", {
  children: _propTypes.default.node.isRequired,
  listRef: _propTypes.default.any,
  className: _propTypes.default.string,
  testId: _propTypes.default.string,
  border: _propTypes.default.oneOf(['top', 'bottom']),
  maxHeight: _propTypes.default.number
});
(0, _defineProperty2.default)(DropdownList, "defaultProps", defaultProps);
var _default = DropdownList;
exports.default = _default;
