// @contentful/forma-36-react-components/dist/components/Typography/Typography/Typography.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.Typography = exports.TypographyContext = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _react = _interopRequireWildcard(require("react"));

var _classnames = _interopRequireDefault(require("classnames"));

var styles = {
  "Typography": "Typography__Typography___1ZUfE"
};
var defaultConfiguration = {
  displayText: {
    spacing: 'l'
  },
  displayTextLarge: {
    spacing: 'xl'
  },
  heading: {
    spacing: 'm'
  },
  paragraph: {
    spacing: 'm'
  },
  sectionHeading: {
    spacing: 'l'
  },
  subheading: {
    spacing: 'm'
  }
};
var defaultProps = {
  testId: 'cf-ui-text-container'
};

var TypographyContext = _react.default.createContext({});

exports.TypographyContext = TypographyContext;

var Typography =
/*#__PURE__*/
function (_Component) {
  (0, _inherits2.default)(Typography, _Component);

  function Typography() {
    (0, _classCallCheck2.default)(this, Typography);
    return (0, _possibleConstructorReturn2.default)(this, (0, _getPrototypeOf2.default)(Typography).apply(this, arguments));
  }

  (0, _createClass2.default)(Typography, [{
    key: "render",
    value: function render() {
      var _this$props = this.props,
          className = _this$props.className,
          children = _this$props.children,
          testId = _this$props.testId,
          otherProps = (0, _objectWithoutProperties2.default)(_this$props, ["className", "children", "testId"]);
      var classNames = (0, _classnames.default)(styles.Typography, className);
      return _react.default.createElement(TypographyContext.Provider, {
        value: defaultConfiguration
      }, _react.default.createElement("div", (0, _extends2.default)({}, otherProps, {
        className: classNames,
        "data-test-id": testId
      }), children));
    }
  }]);
  return Typography;
}(_react.Component);

exports.Typography = Typography;
Typography.displayName = "Typography";
(0, _defineProperty2.default)(Typography, "propTypes", {
  className: _propTypes.default.string,
  children: _propTypes.default.node,
  style: _propTypes.default.any,
  testId: _propTypes.default.string
});
(0, _defineProperty2.default)(Typography, "defaultProps", defaultProps);
var _default = Typography;
exports.default = _default;
