// @contentful/forma-36-react-components/dist/components/Card/Card/Card.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.Card = void 0;

var _objectSpread2 = _interopRequireDefault(require("@babel/runtime/helpers/objectSpread"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf3 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _assertThisInitialized2 = _interopRequireDefault(require("@babel/runtime/helpers/assertThisInitialized"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _react = _interopRequireWildcard(require("react"));

var _classnames = _interopRequireDefault(require("classnames"));

var styles = {
  "Card": "Card__Card___xXJsT",
  "Card--padding-default": "Card__Card--padding-default___b6Brl",
  "Card--padding-large": "Card__Card--padding-large___3Fgal",
  "Card--padding-none": "Card__Card--padding-none___1Jrxi",
  "Card--is-interactive": "Card__Card--is-interactive___dMQtf",
  "Card--is-selected": "Card__Card--is-selected___3zM50"
};
var defaultProps = {
  padding: 'default',
  testId: 'cf-ui-card',
  selected: false
};

var Card =
/*#__PURE__*/
function (_Component) {
  (0, _inherits2.default)(Card, _Component);

  function Card() {
    var _getPrototypeOf2;

    var _this;

    (0, _classCallCheck2.default)(this, Card);

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this = (0, _possibleConstructorReturn2.default)(this, (_getPrototypeOf2 = (0, _getPrototypeOf3.default)(Card)).call.apply(_getPrototypeOf2, [this].concat(args)));
    (0, _defineProperty2.default)((0, _assertThisInitialized2.default)(_this), "handleClick", function (event) {
      if (!_this.props.onClick) return;

      _this.props.onClick(event);
    });
    return _this;
  }

  (0, _createClass2.default)(Card, [{
    key: "render",
    value: function render() {
      var _cn;

      var _this$props = this.props,
          className = _this$props.className,
          testId = _this$props.testId,
          children = _this$props.children,
          href = _this$props.href,
          onClick = _this$props.onClick,
          padding = _this$props.padding,
          selected = _this$props.selected,
          otherProps = (0, _objectWithoutProperties2.default)(_this$props, ["className", "testId", "children", "href", "onClick", "padding", "selected"]);
      var classNames = (0, _classnames.default)(styles.Card, className, (_cn = {}, (0, _defineProperty2.default)(_cn, styles["Card--padding-".concat(padding)], padding), (0, _defineProperty2.default)(_cn, styles['Card--is-interactive'], onClick || href), (0, _defineProperty2.default)(_cn, styles['Card--is-selected'], selected), _cn));
      var Element = href ? 'a' : 'div';
      return _react.default.createElement(Element, (0, _objectSpread2.default)({
        href: href,
        className: classNames,
        onClick: this.handleClick,
        'data-test-id': testId
      }, otherProps), children);
    }
  }]);
  return Card;
}(_react.Component);

exports.Card = Card;
Card.displayName = "Card";
(0, _defineProperty2.default)(Card, "propTypes", {
  href: _propTypes.default.string,
  onClick: _propTypes.default.func,
  padding: _propTypes.default.oneOf(['default', 'large', 'none']),
  selected: _propTypes.default.bool,
  title: _propTypes.default.string,
  style: _propTypes.default.any,
  className: _propTypes.default.string,
  testId: _propTypes.default.string,
  children: _propTypes.default.node.isRequired
});
(0, _defineProperty2.default)(Card, "defaultProps", defaultProps);
var _default = Card;
exports.default = _default;
