// @contentful/forma-36-react-components/dist/components/Dropdown/DropdownListItem/DropdownListItem.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.DropdownListItem = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf3 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _assertThisInitialized2 = _interopRequireDefault(require("@babel/runtime/helpers/assertThisInitialized"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _react = _interopRequireWildcard(require("react"));

var _classnames = _interopRequireDefault(require("classnames"));

var _TabFocusTrap = _interopRequireDefault(require("../../TabFocusTrap/TabFocusTrap"));

var styles = {
  "DropdownListItem": "DropdownListItem__DropdownListItem___LOUnP",
  "DropdownListItem__button": "DropdownListItem__DropdownListItem__button___1Po6h a11y__focus-outline--default___2hwb1",
  "DropdownListItem__submenu-toggle": "DropdownListItem__DropdownListItem__submenu-toggle___1SVw1",
  "DropdownListItem__button__inner-wrapper": "DropdownListItem__DropdownListItem__button__inner-wrapper___LFlkP",
  "DropdownListItem--disabled": "DropdownListItem__DropdownListItem--disabled___1txXv",
  "DropdownListItem--active": "DropdownListItem__DropdownListItem--active___21eet",
  "DropdownListItem--title": "DropdownListItem__DropdownListItem--title___CyVKB"
};
var defaultProps = {
  testId: 'cf-ui-dropdown-list-item',
  isDisabled: false,
  isActive: false,
  isTitle: false
};

var DropdownListItem =
/*#__PURE__*/
function (_Component) {
  (0, _inherits2.default)(DropdownListItem, _Component);

  function DropdownListItem() {
    var _getPrototypeOf2;

    var _this;

    (0, _classCallCheck2.default)(this, DropdownListItem);

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this = (0, _possibleConstructorReturn2.default)(this, (_getPrototypeOf2 = (0, _getPrototypeOf3.default)(DropdownListItem)).call.apply(_getPrototypeOf2, [this].concat(args)));
    (0, _defineProperty2.default)((0, _assertThisInitialized2.default)(_this), "renderSubmenuToggle", function () {
      var _this$props = _this.props,
          onClick = _this$props.onClick,
          onEnter = _this$props.onEnter,
          onLeave = _this$props.onLeave,
          onFocus = _this$props.onFocus,
          otherProps = (0, _objectWithoutProperties2.default)(_this$props, ["onClick", "onEnter", "onLeave", "onFocus"]);
      return _react.default.createElement(_react.default.Fragment, null, _react.default.createElement("button", (0, _extends2.default)({
        type: "button",
        "data-test-id": "cf-ui-dropdown-submenu-toggle",
        className: styles['DropdownListItem__button'],
        onClick: onClick,
        onMouseEnter: onEnter,
        onFocus: onFocus,
        onMouseLeave: onLeave
      }, otherProps), _react.default.createElement(_TabFocusTrap.default, {
        className: styles['DropdownListItem__button__inner-wrapper']
      }, _this.props.submenuToggleLabel)), _this.props.children);
    });
    (0, _defineProperty2.default)((0, _assertThisInitialized2.default)(_this), "renderListItem", function () {
      var _this$props2 = _this.props,
          onClick = _this$props2.onClick,
          onMouseDown = _this$props2.onMouseDown,
          href = _this$props2.href,
          isDisabled = _this$props2.isDisabled,
          children = _this$props2.children,
          isTitle = _this$props2.isTitle,
          isActive = _this$props2.isActive,
          testId = _this$props2.testId,
          listItemRef = _this$props2.listItemRef,
          otherProps = (0, _objectWithoutProperties2.default)(_this$props2, ["onClick", "onMouseDown", "href", "isDisabled", "children", "isTitle", "isActive", "testId", "listItemRef"]);
      var isClickable = onClick || onMouseDown || href;

      if (isClickable) {
        var Element = href ? 'a' : 'button';
        var buttonProps = {
          disabled: isDisabled,
          'aria-disabled': isDisabled
        };
        var linkProps = {
          href: href
        };
        return _react.default.createElement(Element, (0, _extends2.default)({
          type: "button",
          onClick: function (_onClick) {
            function onClick(_x) {
              return _onClick.apply(this, arguments);
            }

            onClick.toString = function () {
              return _onClick.toString();
            };

            return onClick;
          }(function (e) {
            if (!isDisabled && onClick) {
              onClick(e);
            }
          }),
          onMouseDown: function (_onMouseDown) {
            function onMouseDown(_x2) {
              return _onMouseDown.apply(this, arguments);
            }

            onMouseDown.toString = function () {
              return _onMouseDown.toString();
            };

            return onMouseDown;
          }(function (e) {
            if (!isDisabled && onMouseDown) {
              onMouseDown(e);
            }
          })
        }, href ? linkProps : buttonProps, otherProps, {
          "data-test-id": "cf-ui-dropdown-list-item-button",
          className: styles['DropdownListItem__button']
        }), _react.default.createElement(_TabFocusTrap.default, {
          className: styles['DropdownListItem__button__inner-wrapper']
        }, children));
      }

      return _react.default.createElement("span", otherProps, children);
    });
    return _this;
  }

  (0, _createClass2.default)(DropdownListItem, [{
    key: "render",
    value: function render() {
      var _cn;

      var _this$props3 = this.props,
          className = _this$props3.className,
          isDisabled = _this$props3.isDisabled,
          testId = _this$props3.testId,
          listItemRef = _this$props3.listItemRef,
          isActive = _this$props3.isActive,
          onClick = _this$props3.onClick,
          onMouseDown = _this$props3.onMouseDown,
          href = _this$props3.href,
          submenuToggleLabel = _this$props3.submenuToggleLabel,
          isTitle = _this$props3.isTitle;
      var classNames = (0, _classnames.default)(styles['DropdownListItem'], className, (_cn = {}, (0, _defineProperty2.default)(_cn, styles['DropdownListItem__submenu-toggle'], submenuToggleLabel || onClick || onMouseDown || href), (0, _defineProperty2.default)(_cn, styles['DropdownListItem--disabled'], isDisabled), (0, _defineProperty2.default)(_cn, styles['DropdownListItem--active'], isActive), (0, _defineProperty2.default)(_cn, styles['DropdownListItem--title'], isTitle), _cn));
      return _react.default.createElement("li", {
        className: classNames,
        "data-test-id": testId,
        ref: listItemRef,
        role: "menuitem"
      }, submenuToggleLabel ? this.renderSubmenuToggle() : this.renderListItem());
    }
  }]);
  return DropdownListItem;
}(_react.Component);

exports.DropdownListItem = DropdownListItem;
DropdownListItem.displayName = "DropdownListItem";
(0, _defineProperty2.default)(DropdownListItem, "propTypes", {
  isDisabled: _propTypes.default.bool.isRequired,
  listItemRef: _propTypes.default.any,
  isActive: _propTypes.default.bool.isRequired,
  isTitle: _propTypes.default.bool.isRequired,
  children: _propTypes.default.node.isRequired,
  onClick: _propTypes.default.func,
  onMouseDown: _propTypes.default.func,
  submenuToggleLabel: _propTypes.default.string,
  href: _propTypes.default.string,
  onFocus: _propTypes.default.func,
  onLeave: _propTypes.default.func,
  onEnter: _propTypes.default.func,
  className: _propTypes.default.string,
  testId: _propTypes.default.string
});
(0, _defineProperty2.default)(DropdownListItem, "defaultProps", defaultProps);
var _default = DropdownListItem;
exports.default = _default;
