// @contentful/forma-36-react-components/dist/components/Table/TableHead/TableHead.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.TableHead = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectSpread2 = _interopRequireDefault(require("@babel/runtime/helpers/objectSpread"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _react = _interopRequireWildcard(require("react"));

var _classnames = _interopRequireDefault(require("classnames"));

var _TableCell = require("../TableCell");

var styles = {
  "TableHead--sticky": "TableHead__TableHead--sticky___1YzE0"
};
var defaultProps = {
  isSticky: false,
  testId: 'cf-ui-table-head'
};

var TableHead =
/*#__PURE__*/
function (_Component) {
  (0, _inherits2.default)(TableHead, _Component);

  function TableHead() {
    (0, _classCallCheck2.default)(this, TableHead);
    return (0, _possibleConstructorReturn2.default)(this, (0, _getPrototypeOf2.default)(TableHead).apply(this, arguments));
  }

  (0, _createClass2.default)(TableHead, [{
    key: "render",
    value: function render() {
      var _this$props = this.props,
          className = _this$props.className,
          testId = _this$props.testId,
          offsetTop = _this$props.offsetTop,
          isSticky = _this$props.isSticky,
          children = _this$props.children,
          otherProps = (0, _objectWithoutProperties2.default)(_this$props, ["className", "testId", "offsetTop", "isSticky", "children"]);
      var classNames = (0, _classnames.default)(className, (0, _defineProperty2.default)({}, styles["TableHead--sticky"], isSticky));
      return _react.default.createElement(_TableCell.TableCellContext.Provider, {
        value: (0, _objectSpread2.default)({}, _TableCell.contextOptions.head, {
          offsetTop: offsetTop || 0
        })
      }, _react.default.createElement("thead", (0, _extends2.default)({
        className: classNames,
        "data-test-id": testId
      }, otherProps), children));
    }
  }]);
  return TableHead;
}(_react.Component);

exports.TableHead = TableHead;
TableHead.displayName = "TableHead";
(0, _defineProperty2.default)(TableHead, "propTypes", {
  isSticky: _propTypes.default.bool,
  offsetTop: _propTypes.default.oneOfType([_propTypes.default.number, _propTypes.default.string]),
  className: _propTypes.default.string,
  testId: _propTypes.default.string,
  style: _propTypes.default.any,
  children: _propTypes.default.node.isRequired
});
(0, _defineProperty2.default)(TableHead, "defaultProps", defaultProps);
var _default = TableHead;
exports.default = _default;
