// @contentful/forma-36-react-components/dist/components/List/List/List.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.List = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _react = _interopRequireWildcard(require("react"));

var _classnames = _interopRequireDefault(require("classnames"));

var styles = {
  "List": "List__List___1awv4"
};
var defaultProps = {
  testId: 'cf-ui-list'
};

var List =
/*#__PURE__*/
function (_Component) {
  (0, _inherits2.default)(List, _Component);

  function List() {
    (0, _classCallCheck2.default)(this, List);
    return (0, _possibleConstructorReturn2.default)(this, (0, _getPrototypeOf2.default)(List).apply(this, arguments));
  }

  (0, _createClass2.default)(List, [{
    key: "render",
    value: function render() {
      var _this$props = this.props,
          className = _this$props.className,
          children = _this$props.children,
          testId = _this$props.testId,
          otherProps = (0, _objectWithoutProperties2.default)(_this$props, ["className", "children", "testId"]);
      var classNames = (0, _classnames.default)(styles['List'], className);
      return _react.default.createElement("ul", (0, _extends2.default)({}, otherProps, {
        className: classNames,
        "data-test-id": testId
      }), children);
    }
  }]);
  return List;
}(_react.Component);

exports.List = List;
List.displayName = "List";
(0, _defineProperty2.default)(List, "propTypes", {
  className: _propTypes.default.string,
  children: _propTypes.default.node.isRequired,
  style: _propTypes.default.any,
  testId: _propTypes.default.string
});
(0, _defineProperty2.default)(List, "defaultProps", defaultProps);
var _default = List;
exports.default = _default;
