// @contentful/forma-36-react-components/dist/components/Skeleton/SkeletonDisplayText/SkeletonDisplayText.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.SkeletonDisplayText = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _react = _interopRequireWildcard(require("react"));

var _SkeletonText = _interopRequireDefault(require("../SkeletonText"));

var defaultProps = {
  numberOfLines: 1,
  width: 100,
  offsetTop: 0,
  offsetLeft: 0,
  lineHeight: 21,
  marginBottom: 20
};

var SkeletonDisplayText =
/*#__PURE__*/
function (_Component) {
  (0, _inherits2.default)(SkeletonDisplayText, _Component);

  function SkeletonDisplayText() {
    (0, _classCallCheck2.default)(this, SkeletonDisplayText);
    return (0, _possibleConstructorReturn2.default)(this, (0, _getPrototypeOf2.default)(SkeletonDisplayText).apply(this, arguments));
  }

  (0, _createClass2.default)(SkeletonDisplayText, [{
    key: "render",
    value: function render() {
      return _react.default.createElement(_SkeletonText.default, this.props);
    }
  }]);
  return SkeletonDisplayText;
}(_react.Component);

exports.SkeletonDisplayText = SkeletonDisplayText;
SkeletonDisplayText.displayName = "SkeletonDisplayText";
(0, _defineProperty2.default)(SkeletonDisplayText, "defaultProps", defaultProps);
var _default = SkeletonDisplayText;
exports.default = _default;
