// @contentful/forma-36-react-components/dist/components/Notification/NotificationItem.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.NotificationItem = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _react = _interopRequireWildcard(require("react"));

var _classnames = _interopRequireDefault(require("classnames"));

var _IconButton = _interopRequireDefault(require("../IconButton"));

var _Icon = _interopRequireDefault(require("../Icon"));

var styles = {
  "NotificationItem": "NotificationItem__NotificationItem___22iZo",
  "NotificationItem--success": "NotificationItem__NotificationItem--success___1CO7O",
  "NotificationItem--error": "NotificationItem__NotificationItem--error___19wsc",
  "NotificationItem--warning": "NotificationItem__NotificationItem--warning___Bi-Iv",
  "NotificationItem__intent": "NotificationItem__NotificationItem__intent___cC494 helpers__sr-only___3Kv3z",
  "NotificationItem__icon": "NotificationItem__NotificationItem__icon___3gdKj",
  "NotificationItem__text": "NotificationItem__NotificationItem__text___1-1Up",
  "NotificationItem__dismiss": "NotificationItem__NotificationItem__dismiss___1Z6Df"
};
var defaultProps = {
  testId: 'cf-ui-notification',
  intent: 'success',
  hasCloseButton: true
};

var NotificationItem =
/*#__PURE__*/
function (_Component) {
  (0, _inherits2.default)(NotificationItem, _Component);

  function NotificationItem() {
    (0, _classCallCheck2.default)(this, NotificationItem);
    return (0, _possibleConstructorReturn2.default)(this, (0, _getPrototypeOf2.default)(NotificationItem).apply(this, arguments));
  }

  (0, _createClass2.default)(NotificationItem, [{
    key: "render",
    value: function render() {
      var _this$props = this.props,
          children = _this$props.children,
          testId = _this$props.testId,
          intent = _this$props.intent,
          onClose = _this$props.onClose,
          hasCloseButton = _this$props.hasCloseButton;
      var classes = (0, _classnames.default)(styles.NotificationItem, (0, _defineProperty2.default)({}, styles["NotificationItem--".concat(intent)], true));
      return _react.default.createElement("div", {
        className: classes,
        "data-test-id": testId,
        "data-intent": intent,
        role: "alert",
        "aria-live": intent === 'success' ? 'polite' : 'assertive'
      }, _react.default.createElement("div", {
        className: styles.NotificationItem__intent
      }, intent), _react.default.createElement("div", {
        className: styles.NotificationItem__icon,
        "aria-hidden": "true"
      }, _react.default.createElement(_Icon.default, {
        icon: intent === 'success' ? 'CheckCircle' : 'Warning',
        color: "white"
      })), _react.default.createElement("div", {
        className: styles.NotificationItem__text
      }, children), hasCloseButton && _react.default.createElement(_IconButton.default, {
        buttonType: "white",
        iconProps: {
          icon: 'Close'
        },
        onClick: function onClick() {
          if (onClose) {
            onClose();
          }
        },
        testId: "cf-ui-notification-close",
        label: "Dismiss",
        className: styles.NotificationItem__dismiss
      }));
    }
  }]);
  return NotificationItem;
}(_react.Component);

exports.NotificationItem = NotificationItem;
NotificationItem.displayName = "NotificationItem";
(0, _defineProperty2.default)(NotificationItem, "propTypes", {
  intent: _propTypes.default.oneOf(['success', 'error', 'warning']).isRequired,
  hasCloseButton: _propTypes.default.bool,
  onClose: _propTypes.default.any,
  testId: _propTypes.default.string,
  children: _propTypes.default.node.isRequired
});
(0, _defineProperty2.default)(NotificationItem, "defaultProps", defaultProps);
var _default = NotificationItem;
exports.default = _default;
