// @contentful/forma-36-react-components/dist/components/Notification/NotificationsManager.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.NotificationsManager = void 0;

var _toConsumableArray2 = _interopRequireDefault(require("@babel/runtime/helpers/toConsumableArray"));

var _objectSpread2 = _interopRequireDefault(require("@babel/runtime/helpers/objectSpread"));

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _assertThisInitialized2 = _interopRequireDefault(require("@babel/runtime/helpers/assertThisInitialized"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames = _interopRequireDefault(require("classnames"));

var _NotificationItemContainer = _interopRequireDefault(require("./NotificationItemContainer"));

var styles = {
  "NotificationsManager": "NotificationsManager__NotificationsManager___1uvY2",
  "NotificationsManager__container": "NotificationsManager__NotificationsManager__container___3U0e9",
  "NotificationsManager--top": "NotificationsManager__NotificationsManager--top___aRv7j"
};
var uniqueId = 0;

var getUniqueId = function getUniqueId() {
  uniqueId += 1;
  return uniqueId;
};

var NotificationsManager =
/*#__PURE__*/
function (_PureComponent) {
  (0, _inherits2.default)(NotificationsManager, _PureComponent);

  function NotificationsManager(props) {
    var _this;

    (0, _classCallCheck2.default)(this, NotificationsManager);
    _this = (0, _possibleConstructorReturn2.default)(this, (0, _getPrototypeOf2.default)(NotificationsManager).call(this, props));
    (0, _defineProperty2.default)((0, _assertThisInitialized2.default)(_this), "setPosition", function (position, params) {
      if (position === 'bottom' || position === 'top') {
        var _positionOffset = params && params.offset ? params.offset : 20;

        _this.setState({
          position: position,
          positionOffset: _positionOffset
        });
      }
    });
    (0, _defineProperty2.default)((0, _assertThisInitialized2.default)(_this), "setDuration", function (duration) {
      _this.setState({
        duration: duration
      });
    });
    (0, _defineProperty2.default)((0, _assertThisInitialized2.default)(_this), "close", function (id) {
      _this.setState(function (state) {
        return {
          items: state.items.map(function (item) {
            if (item.id !== id) {
              return item;
            }

            return (0, _objectSpread2.default)({}, item, {
              isShown: false
            });
          })
        };
      });
    });
    (0, _defineProperty2.default)((0, _assertThisInitialized2.default)(_this), "closeAndDelete", function (id) {
      _this.setState(function (state) {
        return {
          items: state.items.filter(function (item) {
            return item.id !== id;
          })
        };
      });
    });
    (0, _defineProperty2.default)((0, _assertThisInitialized2.default)(_this), "closeAll", function () {
      _this.setState(function (state) {
        return {
          items: state.items.map(function (item) {
            return (0, _objectSpread2.default)({}, item, {
              isShown: false
            });
          })
        };
      });
    });
    (0, _defineProperty2.default)((0, _assertThisInitialized2.default)(_this), "show", function (text, settings) {
      var duration = settings && settings.duration ? settings.duration : _this.state.duration;
      var intent = settings && settings.intent ? settings.intent : 'success';
      var canClose = settings && typeof settings.canClose !== 'undefined' ? settings.canClose : true;
      var notificationId = settings && settings.id ? settings.id : getUniqueId();
      var notification = {
        id: notificationId,
        text: text,
        close: function close() {
          return _this.closeAndDelete(notificationId);
        },
        duration: duration,
        canClose: canClose,
        isShown: true,
        intent: intent
      };

      var alreadyThere = _this.state.items.find(function (item) {
        return item.id === notification.id;
      });

      if (alreadyThere) {
        return alreadyThere;
      }

      _this.setState(function (state) {
        if (state.position === 'top') {
          return (0, _objectSpread2.default)({}, state, {
            items: [notification].concat((0, _toConsumableArray2.default)(state.items))
          });
        }

        return (0, _objectSpread2.default)({}, state, {
          items: [].concat((0, _toConsumableArray2.default)(state.items), [notification])
        });
      });

      return notification;
    });
    _this.state = {
      items: [],
      position: 'bottom',
      positionOffset: 20,
      duration: 6000
    };

    _this.props.register('close', _this.close);

    _this.props.register('show', _this.show);

    _this.props.register('closeAll', _this.closeAll);

    _this.props.register('setPosition', _this.setPosition);

    _this.props.register('setDuration', _this.setDuration);

    return _this;
  }

  (0, _createClass2.default)(NotificationsManager, [{
    key: "render",
    value: function render() {
      return _react.default.createElement("div", {
        "data-test-id": "cf-notification-container",
        className: (0, _classnames.default)(styles.NotificationsManager, (0, _defineProperty2.default)({}, styles["NotificationsManager--top"], this.state.position === 'top')),
        style: (0, _defineProperty2.default)({}, this.state.position, this.state.positionOffset)
      }, _react.default.createElement("div", {
        className: styles.NotificationsManager__container
      }, this.state.items.map(function (item) {
        return _react.default.createElement(_NotificationItemContainer.default, {
          intent: item.intent,
          duration: item.duration,
          key: item.id,
          hasCloseButton: item.canClose,
          onClose: item.close,
          isShown: item.isShown
        }, item.text);
      })));
    }
  }]);
  return NotificationsManager;
}(_react.PureComponent);

exports.NotificationsManager = NotificationsManager;
NotificationsManager.displayName = "NotificationsManager";
(0, _defineProperty2.default)(NotificationsManager, "propTypes", {
  register: _propTypes.default.func.isRequired
});
var _default = NotificationsManager;
exports.default = _default;
