// @contentful/forma-36-react-components/dist/components/Notification/NotificationItemContainer.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.NotificationItemContainer = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf3 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _assertThisInitialized2 = _interopRequireDefault(require("@babel/runtime/helpers/assertThisInitialized"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _react = _interopRequireWildcard(require("react"));

var _reactAnimateHeight = _interopRequireDefault(require("react-animate-height"));

var _NotificationItem = _interopRequireDefault(require("./NotificationItem"));

var defaultProps = {
  isShown: false
};

var NotificationItemContainer =
/*#__PURE__*/
function (_Component) {
  (0, _inherits2.default)(NotificationItemContainer, _Component);

  function NotificationItemContainer() {
    var _getPrototypeOf2;

    var _this;

    (0, _classCallCheck2.default)(this, NotificationItemContainer);

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this = (0, _possibleConstructorReturn2.default)(this, (_getPrototypeOf2 = (0, _getPrototypeOf3.default)(NotificationItemContainer)).call.apply(_getPrototypeOf2, [this].concat(args)));
    (0, _defineProperty2.default)((0, _assertThisInitialized2.default)(_this), "timer", null);
    (0, _defineProperty2.default)((0, _assertThisInitialized2.default)(_this), "state", {
      isShown: false
    });
    (0, _defineProperty2.default)((0, _assertThisInitialized2.default)(_this), "startTimer", function () {
      if (_this.props.duration) {
        _this.timer = setTimeout(function () {
          _this.close();
        }, _this.props.duration);
      }
    });
    (0, _defineProperty2.default)((0, _assertThisInitialized2.default)(_this), "stopTimer", function () {
      if (_this.timer) {
        clearTimeout(_this.timer);
        _this.timer = null;
      }
    });
    (0, _defineProperty2.default)((0, _assertThisInitialized2.default)(_this), "close", function () {
      _this.stopTimer();

      _this.setState({
        isShown: false
      });
    });
    (0, _defineProperty2.default)((0, _assertThisInitialized2.default)(_this), "handleMouseEnter", function () {
      _this.stopTimer();
    });
    (0, _defineProperty2.default)((0, _assertThisInitialized2.default)(_this), "handleMouseLeave", function () {
      _this.startTimer();
    });
    return _this;
  }

  (0, _createClass2.default)(NotificationItemContainer, [{
    key: "componentDidMount",
    value: function componentDidMount() {
      this.startTimer(); // eslint-disable-next-line

      this.setState({
        isShown: true
      });
    }
  }, {
    key: "componentDidUpdate",
    value: function componentDidUpdate(prevProps) {
      if (prevProps.isShown !== this.props.isShown) {
        // eslint-disable-next-line
        this.setState({
          isShown: this.props.isShown
        });
      }
    }
  }, {
    key: "componentWillUnmount",
    value: function componentWillUnmount() {
      this.stopTimer();
    }
  }, {
    key: "render",
    value: function render() {
      var _this2 = this;

      var _this$props = this.props,
          duration = _this$props.duration,
          rest = (0, _objectWithoutProperties2.default)(_this$props, ["duration"]);
      return _react.default.createElement(_reactAnimateHeight.default, {
        duration: 200,
        height: this.state.isShown ? 'auto' : 0,
        easing: "ease-in-out",
        animateOpacity: true,
        onAnimationEnd: function onAnimationEnd() {
          if (_this2.state.isShown === false) {
            if (_this2.props.onClose) {
              _this2.props.onClose();
            }
          }
        }
      }, _react.default.createElement("div", {
        style: {
          pointerEvents: 'all'
        },
        onMouseEnter: this.handleMouseEnter,
        onMouseLeave: this.handleMouseLeave
      }, _react.default.createElement(_NotificationItem.default, (0, _extends2.default)({}, rest, {
        onClose: this.close
      }))));
    }
  }]);
  return NotificationItemContainer;
}(_react.Component);

exports.NotificationItemContainer = NotificationItemContainer;
NotificationItemContainer.displayName = "NotificationItemContainer";
(0, _defineProperty2.default)(NotificationItemContainer, "propTypes", {
  duration: _propTypes.default.number.isRequired,
  isShown: _propTypes.default.bool
});
(0, _defineProperty2.default)(NotificationItemContainer, "defaultProps", defaultProps);
var _default = NotificationItemContainer;
exports.default = _default;
