// @contentful/forma-36-react-components/dist/components/Textarea/Textarea.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.Textarea = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf3 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _assertThisInitialized2 = _interopRequireDefault(require("@babel/runtime/helpers/assertThisInitialized"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _react = _interopRequireWildcard(require("react"));

var _classnames = _interopRequireDefault(require("classnames"));

var styles = {
  "Textarea": "Textarea__Textarea___qcph7",
  "Textarea__textarea": "Textarea__Textarea__textarea___30c64 a11y__focus-border--default___60AXp",
  "Textarea--small": "Textarea__Textarea--small___3duGT",
  "Textarea--medium": "Textarea__Textarea--medium___2ylrR",
  "Textarea--large": "Textarea__Textarea--large___2jIb0",
  "Textarea--full": "Textarea__Textarea--full___1OW4s",
  "Textarea--disabled": "Textarea__Textarea--disabled___2tLQn",
  "Textarea--negative": "Textarea__Textarea--negative___1RyoO"
};
var defaultProps = {
  testId: 'cf-ui-textarea',
  disabled: false,
  required: false,
  width: 'full',
  willBlurOnEsc: true
};

var Textarea =
/*#__PURE__*/
function (_Component) {
  (0, _inherits2.default)(Textarea, _Component);

  function Textarea() {
    var _getPrototypeOf2;

    var _this;

    (0, _classCallCheck2.default)(this, Textarea);

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this = (0, _possibleConstructorReturn2.default)(this, (_getPrototypeOf2 = (0, _getPrototypeOf3.default)(Textarea)).call.apply(_getPrototypeOf2, [this].concat(args)));
    (0, _defineProperty2.default)((0, _assertThisInitialized2.default)(_this), "state", {
      value: _this.props.value
    });
    (0, _defineProperty2.default)((0, _assertThisInitialized2.default)(_this), "handleKeyDown", function (e) {
      var ESC = 27;

      if (_this.props.onKeyDown) {
        _this.props.onKeyDown(e);
      }

      if (e.keyCode === ESC && _this.props.willBlurOnEsc) {
        e.currentTarget.blur();
      }
    });
    return _this;
  }

  (0, _createClass2.default)(Textarea, [{
    key: "UNSAFE_componentWillReceiveProps",
    value: function UNSAFE_componentWillReceiveProps(nextProps) {
      if (this.props.value !== nextProps.value) {
        this.setState({
          value: nextProps.value
        });
      }
    }
  }, {
    key: "render",
    value: function render() {
      var _cn,
          _this2 = this;

      var _this$props = this.props,
          className = _this$props.className,
          testId = _this$props.testId,
          placeholder = _this$props.placeholder,
          maxLength = _this$props.maxLength,
          onChange = _this$props.onChange,
          disabled = _this$props.disabled,
          required = _this$props.required,
          onBlur = _this$props.onBlur,
          error = _this$props.error,
          width = _this$props.width,
          value = _this$props.value,
          name = _this$props.name,
          rows = _this$props.rows,
          id = _this$props.id,
          willBlurOnEsc = _this$props.willBlurOnEsc,
          textareaRef = _this$props.textareaRef,
          otherProps = (0, _objectWithoutProperties2.default)(_this$props, ["className", "testId", "placeholder", "maxLength", "onChange", "disabled", "required", "onBlur", "error", "width", "value", "name", "rows", "id", "willBlurOnEsc", "textareaRef"]);
      var widthClass = "Textarea--".concat(width);
      var classNames = (0, _classnames.default)(styles['Textarea'], className, styles[widthClass], (_cn = {}, (0, _defineProperty2.default)(_cn, styles['Textarea--disabled'], disabled), (0, _defineProperty2.default)(_cn, styles['Textarea--negative'], error), _cn));
      return _react.default.createElement("div", {
        className: classNames
      }, _react.default.createElement("textarea", (0, _extends2.default)({
        "data-test-id": testId,
        "aria-label": name,
        className: styles['Textarea__textarea'],
        id: id,
        rows: rows,
        onBlur: onBlur,
        disabled: disabled,
        placeholder: placeholder,
        name: name,
        onChange: function (_onChange) {
          function onChange(_x) {
            return _onChange.apply(this, arguments);
          }

          onChange.toString = function () {
            return _onChange.toString();
          };

          return onChange;
        }(function (e) {
          if (onChange) {
            onChange(e);
          }

          _this2.setState({
            value: e.target.value
          });
        }),
        maxLength: maxLength,
        value: disabled ? value : this.state && this.state.value,
        onKeyDown: this.handleKeyDown,
        ref: textareaRef
      }, otherProps)));
    }
  }]);
  return Textarea;
}(_react.Component);

exports.Textarea = Textarea;
Textarea.displayName = "Textarea";
(0, _defineProperty2.default)(Textarea, "propTypes", {
  name: _propTypes.default.string,
  id: _propTypes.default.string,
  testId: _propTypes.default.string,
  placeholder: _propTypes.default.string,
  className: _propTypes.default.string,
  width: _propTypes.default.oneOf(['small', 'medium', 'large', 'full']),
  maxLength: _propTypes.default.number,
  required: _propTypes.default.bool,
  disabled: _propTypes.default.bool,
  value: _propTypes.default.string,
  rows: _propTypes.default.number,
  error: _propTypes.default.bool,
  onChange: _propTypes.default.func,
  onBlur: _propTypes.default.func,
  onFocus: _propTypes.default.func,
  onKeyPress: _propTypes.default.func,
  onKeyDown: _propTypes.default.func,
  onKeyUp: _propTypes.default.func,
  willBlurOnEsc: _propTypes.default.bool.isRequired,
  textareaRef: _propTypes.default.any
});
(0, _defineProperty2.default)(Textarea, "defaultProps", defaultProps);
var _default = Textarea;
exports.default = _default;
