// @contentful/forma-36-react-components/dist/components/Card/CardActions/CardActions.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.CardActions = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf3 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _assertThisInitialized2 = _interopRequireDefault(require("@babel/runtime/helpers/assertThisInitialized"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _react = _interopRequireWildcard(require("react"));

var _Dropdown = _interopRequireDefault(require("../../Dropdown/Dropdown"));

var _IconButton = _interopRequireDefault(require("../../IconButton/IconButton"));

var defaultProps = {
  testId: 'cf-ui-card-actions',
  isDisabled: false
};

var CardActions =
/*#__PURE__*/
function (_Component) {
  (0, _inherits2.default)(CardActions, _Component);

  function CardActions() {
    var _getPrototypeOf2;

    var _this;

    (0, _classCallCheck2.default)(this, CardActions);

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this = (0, _possibleConstructorReturn2.default)(this, (_getPrototypeOf2 = (0, _getPrototypeOf3.default)(CardActions)).call.apply(_getPrototypeOf2, [this].concat(args)));
    (0, _defineProperty2.default)((0, _assertThisInitialized2.default)(_this), "state", {
      isDropdownOpen: false
    });
    (0, _defineProperty2.default)((0, _assertThisInitialized2.default)(_this), "handleClick", function (event) {
      _this.setState(function (prevState) {
        return {
          isDropdownOpen: !prevState.isDropdownOpen
        };
      });

      if (_this.props.iconButtonProps && _this.props.iconButtonProps.onClick) {
        event.stopPropagation();
      }
    });
    return _this;
  }

  (0, _createClass2.default)(CardActions, [{
    key: "render",
    value: function render() {
      var _this2 = this;

      var _this$props = this.props,
          className = _this$props.className,
          children = _this$props.children,
          testId = _this$props.testId,
          iconButtonProps = _this$props.iconButtonProps,
          isDisabled = _this$props.isDisabled,
          otherProps = (0, _objectWithoutProperties2.default)(_this$props, ["className", "children", "testId", "iconButtonProps", "isDisabled"]);
      return _react.default.createElement(_Dropdown.default, (0, _extends2.default)({
        onClose: function onClose() {
          _this2.setState({
            isDropdownOpen: false
          });
        },
        position: "bottom-right",
        className: className,
        isOpen: this.state.isDropdownOpen,
        toggleElement: _react.default.createElement(_IconButton.default, (0, _extends2.default)({
          iconProps: {
            icon: 'MoreHorizontal'
          },
          buttonType: "secondary",
          disabled: isDisabled,
          label: "Actions"
        }, iconButtonProps, {
          onClick: function onClick(event) {
            event.preventDefault();

            _this2.handleClick(event);
          }
        }))
      }, otherProps), _react.default.Children.map(children, function (listItems) {
        return _react.default.Children.map(listItems, function (item) {
          // React.Children behaves differently if the object is a Fragment.
          var resolvedChildren = item.type === _react.default.Fragment ? item.props.children : item;

          var enhancedChildren = _react.default.Children.map(resolvedChildren, function (child) {
            return _react.default.cloneElement(child, {
              onClick: function onClick(e) {
                if (child.props.onClick) {
                  child.props.onClick(e);
                }

                _this2.setState({
                  isDropdownOpen: false
                });

                e.stopPropagation();
              }
            });
          });

          return enhancedChildren;
        });
      }));
    }
  }]);
  return CardActions;
}(_react.Component);

exports.CardActions = CardActions;
CardActions.displayName = "CardActions";
(0, _defineProperty2.default)(CardActions, "propTypes", {
  className: _propTypes.default.string,
  iconButtonProps: _propTypes.default.any,
  isDisabled: _propTypes.default.bool,
  children: _propTypes.default.oneOfType([_propTypes.default.element, _propTypes.default.arrayOf(_propTypes.default.element)]).isRequired
});
(0, _defineProperty2.default)(CardActions, "defaultProps", defaultProps);
var _default = CardActions;
exports.default = _default;
